Splash/Splash/View Controllers/LeaderboardViewController.m

```diff
@@ -32,16 +35,43 @@ - (void)viewDidLoad {
     self.leaderboardTableView.sectionHeaderTopPadding = 0;
     [self.refreshControl addTarget:self action:@selector(getLeaderboardData) forControlEvents:UIControlEventValueChanged];
     [self.leaderboardTableView insertSubview:self.refreshControl atIndex:0];
+    self.searchBar.delegate = self;
+    self.tapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyboard)];
+    self.tapGestureRecognizer.delegate = self;
+    [self.view addGestureRecognizer:self.tapGestureRecognizer];
+    
+}
+
+
+- (void) dismissKeyboard
+{
+    //Code to handle the gesture
+    [self.searchBar resignFirstResponder];
 }
 
 - (void) getLeaderboardData {
     PFQuery *query = [PFUser query];
     [query orderByDescending:@"bubblescore"];
     query.limit = 10;
     self.leaderboard = [query findObjects];
+    self.filteredLeaderboard = self.leaderboard;
     [self.refreshControl endRefreshing];
 }
 
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (searchText.length != 0) {
```

Comment:
Can this also handle the case that the input is made of several spaces?

Reply:
yes! It will look for any text that has a specific number of spaces! If nothing matches, then no cells will show up.